jupyter-repl.el

```diff
@@ -962,6 +962,69 @@ buffer to display TEXT."
           (font-lock-prepend-text-property
            beg end 'font-lock-face 'jupyter-repl-traceback)))))))
 
+(defun jupyter-repl-history--match-input (regexp arg)
+  "Return the index of the ARGth REGEXP match.
+Or nil, on failure. If ARG is positive, search backward from the most
+recent history element. If negative, search forward through items
+previously visited during this input session. If ARG is zero, do
+nothing."
+  ;; Adapted from `comint-previous-matching-input-string-position'
+  (let* ((r jupyter-repl-history)
+         (n (ring-length r))
+         (direction (if (> arg 0) +1 -1))
+         (i (if (= direction -1) 0 -1)) ; adjust for initial increment
+         (failed (zerop arg))
+         ;;
+         code)
+    ;; Search ARG times
+    (while (not (or failed (zerop arg)))
+      (while (not (or (setq i (mod (+ i direction) n)
```

Comment:
We don't need this `mod` here since `ring-ref` already does this. I believe we can just do

```emacs-lisp
code (ring-ref r (cl-incf i direction))
```

Reply:
Great, thanks. This also simplifies the post-helper figuring in the main command, since the sign of the returned index now matches that of the numeric arg.